bionic/code_hasher.py

```diff
@@ -85,6 +85,14 @@ def _ingest_raw_prefix_and_bytes(self, type_prefix, obj_bytes=b""):
         self._hash.update(PREFIX_SEPARATOR)
         self._hash.update(obj_bytes)
 
+        # TODO: It is not safe to use just the id of an object since
```

Comment:
Hmmm, nice observation. What if we keep a reference to each object for the duration of the hashing process? That way their lifetimes will all overlap and they should each have a unique memory address.

Reply:
I'll try this out and report back if it works.

Reply:
Yup, this worked for code hasher. I already merged the change as part of #312.